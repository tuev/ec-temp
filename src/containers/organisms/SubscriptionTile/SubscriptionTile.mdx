# Stories mdx for SubscriptionTile

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './SubscriptionTile.stories.tsx'
import SubscriptionTile from './SubscriptionTile.tsx'

<Story name="basic" />