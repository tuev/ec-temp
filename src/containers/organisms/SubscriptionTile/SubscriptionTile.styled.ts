/**
 *
 * Styled for SubscriptionTile
 *
 */

import styled from 'styled-components'
