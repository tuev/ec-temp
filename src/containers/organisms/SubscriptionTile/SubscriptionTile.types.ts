export type SubscriptionTileProps = {
  onSubscribe?: (email: string) => unknown
}
