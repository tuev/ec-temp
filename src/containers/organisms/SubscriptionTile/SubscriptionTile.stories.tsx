/**
 *
 *  Stories for SubscriptionTile
 *
 */

import React from 'react'

import SubscriptionTile from './SubscriptionTile'
export default {
  component: SubscriptionTile,
  includeStories: ['basic'],
  title: 'SubscriptionTile',
}

export const basic = () => {
  return <SubscriptionTile />
}
