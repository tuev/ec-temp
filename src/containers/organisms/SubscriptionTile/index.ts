/**
 *
 * Index for SubscriptionTile
 *
 */

import SubscriptionTile from './SubscriptionTile'

export default SubscriptionTile
