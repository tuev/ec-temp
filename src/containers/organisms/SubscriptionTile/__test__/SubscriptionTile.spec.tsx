/**
 *
 * Tests for SubscriptionTile
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import SubscriptionTile from '..'

describe('App SubscriptionTile', () => {
  it('Test get default SubscriptionTile no params ', () => {
    // test snapshot
    const wrapper = render(<SubscriptionTile />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default SubscriptionTile no event handler ', () => {
    // test snapshot
    const wrapper = render(<SubscriptionTile />)
    expect(wrapper).toMatchSnapshot()

    const subcribeBtn = wrapper.getByTestId('subscribe-btn')

    fireEvent.click(subcribeBtn)
  })

  it('Test get default SubscriptionTile with event handler ', () => {
    let result = ''
    const onSubcribeHandler = jest
      .fn()
      .mockImplementation((value) => (result = value))
    const mockEmail = 'emailtest'
    // test snapshot
    const wrapper = render(<SubscriptionTile onSubscribe={onSubcribeHandler} />)
    expect(wrapper).toMatchSnapshot()

    const emailInput = wrapper.getByTestId('subscribe-email-input')
    fireEvent.change(emailInput, { target: { value: mockEmail } })

    const subcribeBtn = wrapper.getByTestId('subscribe-btn')
    fireEvent.click(subcribeBtn)

    expect(onSubcribeHandler).toBeCalled()
    expect(result).toEqual(mockEmail)
  })
})
