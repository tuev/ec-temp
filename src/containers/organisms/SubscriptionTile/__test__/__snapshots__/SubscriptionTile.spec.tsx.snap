// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App SubscriptionTile Test get default SubscriptionTile no event handler  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #F5F5F5;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 100%;
}

.c4 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 350px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  font-size: 24px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c3 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c5 {
  min-width: 90px !important;
  min-height: 36px !important;
  font-size: 12px;
  border-radius: 24px;
}

@media screen and (min-width:40em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    width: 50%;
  }
}

<div>
      <div
        class="c0"
        display="flex"
        width="100vw"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1 MuiTypography-colorError"
          customvariant="header2"
        >
          10% discount for your subscription
        </p>
        <div
          class="c2"
          width="1,0.5"
        >
          <p
            class="MuiTypography-root c3 MuiTypography-body1 MuiTypography-alignCenter"
            customvariant="header3"
          >
            Carry the day in style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring coloful faux leather trim, this tote offer a roomy iterior
          </p>
        </div>
        <div
          class="c4"
          display="flex"
          width="1,0.5"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="true"
            >
              Your email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                data-testid="subscribe-email-input"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-259 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-261 PrivateNotchedOutline-legendNotched-262"
                >
                  <span>
                    Your email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c5 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="subscribe-btn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Subscribe
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC eXVcOJ"
      display="flex"
      width="100vw"
    >
      <p
        class="MuiTypography-root sc-AykKD eWFSXP MuiTypography-body1 MuiTypography-colorError"
        customvariant="header2"
      >
        10% discount for your subscription
      </p>
      <div
        class="sc-AykKC hSZhAe"
        width="1,0.5"
      >
        <p
          class="MuiTypography-root sc-AykKD bYERao MuiTypography-body1 MuiTypography-alignCenter"
          customvariant="header3"
        >
          Carry the day in style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring coloful faux leather trim, this tote offer a roomy iterior
        </p>
      </div>
      <div
        class="sc-AykKC kOjYBH"
        display="flex"
        width="1,0.5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="true"
          >
            Your email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="subscribe-email-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-259 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-261 PrivateNotchedOutline-legendNotched-262"
              >
                <span>
                  Your email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AykKE ektXmT MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="subscribe-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Subscribe
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App SubscriptionTile Test get default SubscriptionTile no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #F5F5F5;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 100%;
}

.c4 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 350px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  font-size: 24px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c3 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c5 {
  min-width: 90px !important;
  min-height: 36px !important;
  font-size: 12px;
  border-radius: 24px;
}

@media screen and (min-width:40em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    width: 50%;
  }
}

<body>
    <div>
      <div
        class="c0"
        display="flex"
        width="100vw"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1 MuiTypography-colorError"
          customvariant="header2"
        >
          10% discount for your subscription
        </p>
        <div
          class="c2"
          width="1,0.5"
        >
          <p
            class="MuiTypography-root c3 MuiTypography-body1 MuiTypography-alignCenter"
            customvariant="header3"
          >
            Carry the day in style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring coloful faux leather trim, this tote offer a roomy iterior
          </p>
        </div>
        <div
          class="c4"
          display="flex"
          width="1,0.5"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="true"
            >
              Your email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                data-testid="subscribe-email-input"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-107 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-109 PrivateNotchedOutline-legendNotched-110"
                >
                  <span>
                    Your email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c5 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="subscribe-btn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Subscribe
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC eXVcOJ"
      display="flex"
      width="100vw"
    >
      <p
        class="MuiTypography-root sc-AykKD eWFSXP MuiTypography-body1 MuiTypography-colorError"
        customvariant="header2"
      >
        10% discount for your subscription
      </p>
      <div
        class="sc-AykKC hSZhAe"
        width="1,0.5"
      >
        <p
          class="MuiTypography-root sc-AykKD bYERao MuiTypography-body1 MuiTypography-alignCenter"
          customvariant="header3"
        >
          Carry the day in style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring coloful faux leather trim, this tote offer a roomy iterior
        </p>
      </div>
      <div
        class="sc-AykKC kOjYBH"
        display="flex"
        width="1,0.5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="true"
          >
            Your email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="subscribe-email-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-107 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-109 PrivateNotchedOutline-legendNotched-110"
              >
                <span>
                  Your email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AykKE ektXmT MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="subscribe-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Subscribe
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App SubscriptionTile Test get default SubscriptionTile with event handler  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #F5F5F5;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 100%;
}

.c4 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 350px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  font-size: 24px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c3 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Montserrat Alternates';
  font-weight: bold;
  color: #2E2E2E;
}

.c5 {
  min-width: 90px !important;
  min-height: 36px !important;
  font-size: 12px;
  border-radius: 24px;
}

@media screen and (min-width:40em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    width: 50%;
  }
}

<div>
      <div
        class="c0"
        display="flex"
        width="100vw"
      >
        <p
          class="MuiTypography-root c1 MuiTypography-body1 MuiTypography-colorError"
          customvariant="header2"
        >
          10% discount for your subscription
        </p>
        <div
          class="c2"
          width="1,0.5"
        >
          <p
            class="MuiTypography-root c3 MuiTypography-body1 MuiTypography-alignCenter"
            customvariant="header3"
          >
            Carry the day in style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring coloful faux leather trim, this tote offer a roomy iterior
          </p>
        </div>
        <div
          class="c4"
          display="flex"
          width="1,0.5"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="true"
            >
              Your email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                data-testid="subscribe-email-input"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-411 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-413 PrivateNotchedOutline-legendNotched-414"
                >
                  <span>
                    Your email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c5 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="subscribe-btn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Subscribe
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC eXVcOJ"
      display="flex"
      width="100vw"
    >
      <p
        class="MuiTypography-root sc-AykKD eWFSXP MuiTypography-body1 MuiTypography-colorError"
        customvariant="header2"
      >
        10% discount for your subscription
      </p>
      <div
        class="sc-AykKC hSZhAe"
        width="1,0.5"
      >
        <p
          class="MuiTypography-root sc-AykKD bYERao MuiTypography-body1 MuiTypography-alignCenter"
          customvariant="header3"
        >
          Carry the day in style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring coloful faux leather trim, this tote offer a roomy iterior
        </p>
      </div>
      <div
        class="sc-AykKC kOjYBH"
        display="flex"
        width="1,0.5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="true"
          >
            Your email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="subscribe-email-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-411 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-413 PrivateNotchedOutline-legendNotched-414"
              >
                <span>
                  Your email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AykKE ektXmT MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="subscribe-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Subscribe
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
